Speaker Feedback: Give better information on invalid fields

ryelle opened the pull request:
Return more descriptive errors when validating field values, and add a character counter to each textarea to indicate max length of content.

Fixes #460

### Screenshots

![Screen Shot 2020-04-30 at 4 27 40 PM](https://user-images.githubusercontent.com/541093/80755850-8754da00-8aff-11ea-8093-012948822eef.png)

ryelle committed:
Speaker Feedback: Return more descriptive validation errors

ryelle committed:
Add a character count to each textarea

ryelle committed:
Fix error message logic

coreymckrill commented on public_html/wp-content/plugins/wordcamp-speaker-feedback/assets/js/script.js:
```diff
@@ -72,6 +72,17 @@
 	}
 
+	function characterCounter( event ) {
+		var len = event.target.value.length;
```
This doesn't appear to handle multibyte characters, like emoji, the way I would expect. "💩" counts as two characters instead of one. On the PHP side we're using `mb_strlen` in the `validate_meta_string` callback to ensure that multibytes are only counted as one character, but I'm not sure how best to do that on the JS side.

ryelle replied:
ah, of course it couldn't be this simple. i'll take another look in the morning, but if it still seems too complex I might pull this part out.

coreymckrill replied:
Looks like there might be a solution [here](https://dmitripavlutin.com/what-every-javascript-developer-should-know-about-unicode/#33-string-length), although it might only be compatible with ECMAScript 2015.

Something like

```js
var len = [...event.target.value].length;
```

coreymckrill replied:
Or there's this:
https://mathiasbynens.be/notes/javascript-unicode#accounting-for-astral-symbols

(Sorry, I got intrigued)

ryelle replied:
Done in 34596f713ef7b03d937519a1ab473a93c9178a75

coreymckrill commented on public_html/wp-content/plugins/wordcamp-speaker-feedback/views/form-feedback.php:
```diff
@@ -110,8 +115,14 @@
 			<?php echo $schema[ $key ]['required'] ? 'required' : ''; ?>
 		></textarea>
+		<?php if ( $maxlength_attr ) : ?>
+			<div class="speaker-feedback__field-help">
+				0/<?php echo absint( $schema[ $key ]['attributes']['maxlength'] ); ?>
+			</div>
```
Should this be `aria-hidden`?

ryelle replied:
hm… no, because we want to expose that content to screen readers, but it might need more context. i'll take a look at what other char counters do for a11y.

ryelle replied:
Added a describedby in 6cd11e9d1174bce4f706432801048c81e10b6fc9 — the counter stays available to all users (since some sighted users use screen readers), and now has more context with the extra description.

ryelle committed:
Account for "astral symbols" when counting characters, to match PHP count

ryelle committed:
Add a description to the fields for screen reader users

These users might not see the character counter to learn the max length from it, and it may seem disconnected without more context.
